spec/java/util/StringBuilder.main.lsl

```diff
@@ -0,0 +1,1022 @@
+
+        result = self;
+    }
+
+
+    fun *.lastIndexOf (@target self: StringBuilder, str: String): int
+    {
+        result = action DEBUG_DO("this.storage.lastIndexOf(str)");
+    }
+
+
+    fun *.lastIndexOf (@target self: StringBuilder, str: String, fromIndex: int): int
+    {
+        result = action DEBUG_DO("this.storage.lastIndexOf(str, fromIndex)");
+    }
+
+
+    fun *.replace (@target self: StringBuilder, start: int, end: int, str: String): StringBuilder
+    {
+        if (end > this.length)
+            end = this.length;
+
+        _checkRangeSIOOBE(start, end, this.length);
+
+        val strLength: int = action CALL_METHOD(str, "length", []);
+
+        val newLength: int = this.length + strLength - (end - start);
+        var i: int = 0;
+        var strIndex: int = 0;
+        val newStr: array<char> = action ARRAY_NEW("char", newLength);
+
+        action LOOP_FOR(i, 0, newLength, +1, _replace_loop(i, strIndex, start, end, newStr, str, strLength));
+
+        this.storage = action OBJECT_TO_STRING(newStr);
+        this.length = newLength;
+        result = self;
+    }
+
+
+    @Phantom proc _replace_loop (i: int, strIndex: int, start: int, end: int, newStr: array<char>, str: String, strLength: int): void
+    {
+        if (i < start)
+        {
+            newStr[i] = action CALL_METHOD(this.storage, "charAt", [i]);
+        }
+        else if (i < start + strLength)
+        {
+            newStr[i] = action CALL_METHOD(str, "charAt", [strIndex]);
+            strIndex += 1;
+        }
+        else
+        {
+            newStr[i] = action CALL_METHOD(this.storage, "charAt", [end]);
+            end += 1;
+        }
+    }
+
+
+    fun *.reverse (@target self: StringBuilder): StringBuilder
+    {
+        var n: int = this.length - 1;
+        var i: int = (n-1) / 2;
+        var k: int = -1;
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        action LOOP_FOR(i, 0, -1, -1, _reverse_loop(i, k, n, newStr));
+
+        this.storage = action OBJECT_TO_STRING(newStr);
+        result = self;
+    }
+
+
+    @Phantom proc _reverse_loop (i: int, k: int, n: int, newStr: array<char>): void
+    {
+        k = n - i;
+        newStr[k] = action CALL_METHOD(this.storage, "charAt", [i]);
+        newStr[i] = action CALL_METHOD(this.storage, "charAt", [k]);
+    }
+
+
+    fun *.toString (@target self: StringBuilder): String
+    {
+        action NOT_IMPLEMENTED("no final decision");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.capacity (@target self: StringBuilder): int
+    {
+        // This is right realization ?
+        // original: return value.length >> coder;
+        // we will have another result, because we don't know "coder" value;
+        result = this.length;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.ensureCapacity (@target self: StringBuilder, minimumCapacity: int): void
+    {
+
```

Comment:
Add a comment on why this method is empty or add `action DO_NOTHING()` to indicate the intention.

Reply:
Fixed

Reply:
Action `NOT_IMPLEMENTED` is not a correct one here. Consider `DO_NOTHING` instead.